internal/resource/decode.go

```diff
@@ -23,6 +23,22 @@ type DecodedResource[V any, PV interface {
 	Data     PV
 }
 
+func (d *DecodedResource[V, PV]) GetResource() *pbresource.Resource {
```

Comment:
I added these to make it so that you could reference these with an interface type like:
```
type SomeDecodedResource {
  GetResource() *pbresource.Resource
}
```